src/sentry/api/serializers/models/event.py

```diff
@@ -366,12 +369,36 @@ def _get_perf_problem(self, attrs):
         converted_problem["issueType"] = get_group_type_by_type_id(issue_type).slug
         return converted_problem
 
+    def _format_breadcrumb_messages(
+        self, event_data: dict[str, Any], event: Event | GroupEvent, user: User
+    ):
+        breadcrumbs = next(
+            filter(lambda entry: entry["type"] == "breadcrumbs", event_data["entries"]), None
+        )
+
+        if not breadcrumbs or not features.has(
+            "organizations:issue-breadcrumbs-sql-format", event.project.organization, actor=user
+        ):
+            return event_data
+
+        try:
+            for breadcrumb_item in breadcrumbs["data"]["values"]:
+                if breadcrumb_item["category"] in FORMATTED_BREADCRUMB_CATEGORIES:
+                    breadcrumb_item["messageRaw"] = breadcrumb_item["message"]
+                    breadcrumb_item["message"] = sqlparse.format(
+                        breadcrumb_item["message"], reindent_aligned=True
+                    )
+            return event_data
+        except Exception:
```

Comment:
Should we log a message here on exception to make it easier to fix the issue?

Reply:
Good call - this should make a sentry issue now